packages/starlight/__tests__/plugins/config.test.ts

```diff
@@ -120,6 +122,80 @@ describe('validation', () => {
 	});
 });
 
+describe('deprecated `setup` hook', () => {
+	test('supports the legacy `setup` hook in pre v1 versions', async () => {
+		const isPreV1 = pkg.version[0] === '0';
+
+		const pluginResult = runPlugins(
+			{ title: 'Test Docs' },
+			[
+				{
+					name: 'valid-plugin',
+					hooks: { setup() {} },
+				},
+			],
+			createTestPluginContext()
+		);
+
+		if (isPreV1) {
+			await expect(pluginResult).resolves.not.toThrow();
+		} else {
+			await expect(pluginResult).rejects.toThrow();
+		}
+	});
+
+	test('validates plugins have at least a `config:setup` or the deprecated `setup` hook', async () => {
+		await expect(
+			runPlugins(
+				{ title: 'Test Docs' },
+				[{ name: 'invalid-plugin', hooks: {} }],
+				createTestPluginContext()
+			)
+		).rejects.toThrowErrorMatchingInlineSnapshot(`
+			"[AstroUserError]:
+				Invalid plugins config passed to starlight integration
+			Hint:
+				A plugin must define at least a \`config:setup\` hook."
+		`);
+	});
+
+	test('validates plugins does not have both a `config:setup` or the deprecated `setup` hook', async () => {
+		await expect(
+			runPlugins(
+				{ title: 'Test Docs' },
+				[
+					{
+						name: 'invalid-plugin',
+						hooks: {
+							'config:setup'() {},
+							setup() {},
+						},
+					},
+				],
+				createTestPluginContext()
+			)
+		).rejects.toThrowErrorMatchingInlineSnapshot(`
+			"[AstroUserError]:
+				Invalid plugins config passed to starlight integration
+			Hint:
+				A plugin cannot define both a \`config:setup\` and \`setup\` hook. As \`setup\` is deprecated and will be removed in a future version, consider using \`config:setup\` instead."
+		`);
+	});
+
+	test.runIf(pkg.version[0] === '1')(
+		'removes the deprecated `setup` hook in the DocSearch plugin with Starlight v1',
+		async () => {
+			const docSearchPlugin = starlightDocSearch({
+				appId: 'YOUR_APP_ID',
+				apiKey: 'YOUR_SEARCH_API_KEY',
+				indexName: 'YOUR_INDEX_NAME',
+			});
+			expect(docSearchPlugin.hooks.setup).not.toBeDefined();
+			expect(docSearchPlugin.hooks['config:setup']).toBeDefined();
+		}
+	);
+});
```

Comment:
FWIW, I would be happy to just change the DocSearch plugin right away, bumping the minimum supported Starlight version, and release a new minor. Can also happen in a follow-up PR though as we can release a new plugin version after this Starlight minor.

Reply:
Done, one more changeset it is 😅